cozy/ui/book_detail_view.py

```diff
@@ -193,19 +238,27 @@ def _display_external_section(self):
             self.download_switch.set_active(self._view_model.book.offline)
             self.download_switch.handler_unblock_by_func(self._download_switch_changed)
 
-    def _add_disk(self, chapter: Chapter):
+    def _add_disk(self, book_id: int, chapter: Chapter):
+        if book_id != self._view_model.book.id:
+            return
+
         disc_element = DiskElement(chapter.disk)
         self.chapter_box.add(disc_element)
         disc_element.show_all()
+        self._chapters_event.set()
+
+    def _add_chapter(self, book_id: int, chapter: Chapter):
+        if book_id != self._view_model.book.id:
+            return
```

Comment:
Note to self: confirm if we still need these conditions in light of the 443908ecff5b3f269429b4e6b71f95bea06b9519 fix